### ISSUE TITLE
Stack overflow for some binaries

### ISSUE THREAD

**Issue by deetch** [BinaryAnalysisPlatform/bap#1267]
Using bap version 2.3.0-alpha+aacced7 the analysis of some binaries raises a stack overflow. An example of stderr is given here:

```
Failed to build the project:
(("Stack overflow")
  "Raised by primitive operation at file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 754, characters 26-45\
 \nCalled from file \"src/map.ml\", line 1677, characters 11-85\
 \nCalled from file \"lib/knowledge/bap_knowledge.ml\", line 2572, characters 29-274\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 60-63\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
 \nCalled from file \"lib/monads/monads_monad.ml\" (inlined), line 1029, characters 29-34\
 \nCalled from file \"lib/monads/monads_monad.ml\", line 1034, characters 38-42\
[...]
```

In this case the binary was doxygen.
[doxygen.gz](https://github.com/BinaryAnalysisPlatform/bap/files/5867255/doxygen.gz)

**Comment by ivg** [comment]
Can you please also include such information as `bap --version`?

**Comment by ivg** [comment]
While I am waiting for the bactrace to come up on my side, if you have the full trace somewhere nearby, I will really appreciate if you will share it with me.

**Comment by deetch** [comment]
[doxygen-trace.txt](https://github.com/BinaryAnalysisPlatform/bap/files/5868426/doxygen-trace.txt)

**Comment by ivg** [comment]
Thanks, we can reproduce the issue and are working, slowly, on the fix.

**Comment by ivg** [comment]
closed with #1361
